glib/src/value.rs

```diff
@@ -205,6 +205,18 @@ pub unsafe trait FromValue<'a>: Sized {
     unsafe fn from_value(value: &'a Value) -> Self;
 }
 
+/// Similar to `FromValue` but works for owned `Value.
+pub unsafe trait FromOwnedValue: Sized {
+    /// Value type checker.
+    type Checker: ValueTypeChecker;
+
+    /// Get the contained value from a `Value`
+    ///
+    /// # Safety
+    /// `Self::Checker::check()` must be called first and must not fail.
+    unsafe fn from_value(value: &Value) -> Self;
+}
+
```

Comment:
You'll actually have to implement the new trait for `String`, `i32`, etc. (and also put it into the `glib::wrapper!` macros for boxed/shared/boxed-inline/object)